tools/apigee-sackmesser/bin/sackmesser

```diff
@@ -131,5 +129,23 @@ if [[ -z "$token" && (-z "$username"  || -z "$password") ]]; then
     exit 1
 fi
 
+if [[ -z "$token" && "$apiversion" = "apigee" ]]; then
+    if [ -z "$mfa" ]; then MFA=""; else MFA="?mfa_token=$mfa"; fi
+    logdebug "Getting Apigee OAuth2 token for user $username"
+    response=$(curl -fsS -H "Content-Type:application/x-www-form-urlencoded;charset=utf-8" \
+      -H "Accept: application/json;charset=utf-8" \
+      -H "Authorization: Basic ZWRnZWNsaTplZGdlY2xpc2VjcmV0" \
+      -X POST "https://login.apigee.com/oauth/token$MFA" \
+      --data-urlencode "username=$username" \
```

Comment:
im interested in the use of `--data-urlencode` - especially given that this is in the body, not the query param where you would typical encode such things.

have we tested this with a password with characters such as `&`? my instinct is that this would only work with `--data`

Reply:
validated with password containing all kinds of special characters incl. &. Works like a charm.